Scripts/RemediationScripts/Remediate-EnableMicrosoftDefenders.ps1

```diff
@@ -97,6 +97,187 @@ function Setup-Prerequisites
     Write-Host "All required modules are present." -ForegroundColor $([Constants]::MessageType.Update)
 }
 
+function Fetch-API {
+    param (
+        [Parameter(Mandatory=$true)]
+        [string]$Method,
+        
+        [Parameter(Mandatory=$true)]
+        [string]$Uri,
+        
+        [Parameter(Mandatory=$false)]
+        [hashtable]$Body = @{},
+        
+        [Parameter(Mandatory=$false)]
+        [hashtable]$Headers = @{}
+    )
+
+    $cloudEnvironmentResourceManagerUrl = (Get-AzContext).Environment.ResourceManagerUrl
+    $accessToken = Get-AzAccessToken -ResourceUrl $cloudEnvironmentResourceManagerUrl
+    $authHeader = "Bearer " + $accessToken.Token
+    $Headers["Authorization"] = $authHeader
+    $Headers["Content-Type"] = "application/json"
+
+    try {
+        switch ($Method.ToUpper()) {
+            "GET" {
+                $response = Invoke-WebRequest -Uri $Uri -Method Get -Headers $Headers -UseBasicParsing -ErrorAction Stop
+            }
+            "POST" {
+                $jsonBody = $Body | ConvertTo-Json
+                $response = Invoke-WebRequest -Uri $Uri -Method Post -Headers $Headers -Body $jsonBody -UseBasicParsing -ErrorAction Stop
+            }
+            "PUT" {
+                $jsonBody = $Body | ConvertTo-Json
+                $response = Invoke-WebRequest -Uri $Uri -Method Put -Headers $Headers -Body $jsonBody -UseBasicParsing -ErrorAction Stop
+            }
+            "DELETE" {
+                $response = Invoke-WebRequest -Uri $Uri -Method Delete -Headers $Headers -UseBasicParsing -ErrorAction Stop
+            }
+            default {
+                throw "Unsupported HTTP method: $Method"
+            }
+        }
+
+        if ($response.StatusCode -eq 200 -or $response.StatusCode -eq 201) {
+            return $response.Content | ConvertFrom-Json
+        }
+        else {
+            throw "API call failed with status code $($response.StatusCode)"
+        }
+    }
+    catch {
+        Write-Error "Error occurred: $_"
+    }
+}
+
+function Remediate-VirtualMachines {
+    param (
+        [string]$subscriptionId,          # Subscription ID
+        [string]$reqMDCTier,              # Required MDC Tier (e.g., Standard, Free)
+        [string]$currentVulnerabilitySetting, # Current server vulnerability assessment setting (e.g., MdeTvm)
+        [bool]$isWDATPEnabled             # Whether WDATP is already enabled
+    )
+    try {
+        $virtualMachinePricing = Set-AzSecurityPricing -Name "VirtualMachines" -PricingTier $reqMDCTier
+    }
+    catch {
+        Write-Host "Failed to set pricing tier for VirtualMachines"
+        return $_.Exception.Message
+    }
+
+    if ($currentVulnerabilitySetting -ne "MdeTvm") {
```

Comment:
can we use constants for storing "MdeTvm"?

Reply:
similar for other required properties.

Reply:
Added another constant $requiredVulnerabilityAssessmentProvider